### DISCUSSION TITLE
Web Static on S3

### DISCUSSION THREAD

**Discussion by jokourno12** [awsugid/forum#20] [General]
Hai, saya coba buat file HTML dan CSS kemudian saya upload di S3 Bucket. Saya buat menjadi public dan dapat diakses melalui internet. Sepengalaman saya untuk menggunakan DNS maka diperlukan alamat IP dari server. Nah, dalam masalah S3 Bucket saya tidak bisa mendapatkan IP public. Apakah memungkinkan menggunakan URL dari Bucket tersebut untuk mendaftarkan Domain? jika tidak mungkin ada solusi lain?

**Comment by rioastamal** [comment]
@jokourno bisa, yaitu dengan menghubungkan S3 dengan Amazon CloudFront. Saya sendiri menggunakan cara tersebut untuk hosting blog saya di https://teknocerdas.com/

Saya mendokumentasikannya di artikel ini https://teknocerdas.com/serba-serbi/static-wordpress-hosting-pada-infrastruktur-teknocerdas/

Tapi banyak artikel lain yang membahas serupa contohnya:
- https://docs.aws.amazon.com/AmazonS3/latest/userguide/website-hosting-custom-domain-walkthrough.html

**Reply by inthinks** [reply]
ada tutorial lengkapnya gak Pak untuk membangun web teknocerdas tersebut ?
btw untuk setting https dari static web diatas bagaimana caranya ya pak ?

kena biaya berapa perbulan ya Pak untuk semua setup tersebut ?

**Reply by rioastamal** [reply]
@inthinks Untuk tutorial detil secara saya belum buat tapi diinternet banyak. Intinya adalah dari WordPress entah di localhost/EC2/VM manapun generate semua halaman menjadi static HTML. Bisa menggunakan plugin misal https://wordpress.org/plugins/simply-static/ atau sejenis.

Setelah semua file HTML digenerate tinggal sync ke S3 dan letakkan CloudFront didepannya atau pakai Amplify kalau pengen lebih mudah lagi.

Untuk biaya, seingat saya biasanya saya membayar antara $1-$4 dollar per bulan tergantung traffic.

**Comment by jokourno12** [comment]
Thanks coach Rio. Saya kira itu memungkinkan adanya cost optimization dibandingkan penggunaan Wordpress yang memerlukan engine.

**Reply by rioastamal** [reply]
Betul, biasanya saya membayar antara $1-$4 dollar per bulan tergantung traffic.

**Comment by furinongko** [comment]
Kalo di DNS itu ada yg namanya record CNAME, jadi selain menggunakan IP, kita juga bisa CNAME ke url nya S3.

**Reply by rioastamal** [reply]
Tapi tidak bisa HTTPS karena TLS certificate issue jadi hanya HTTP saja yang didukung. Karena itu biasanya perlu CDN didepannya seperti CloudFront yang melakukan SSL termination.
